gdx/src/com/badlogic/gdx/files/FileHandle.java

```diff
@@ -507,12 +532,24 @@ public void writeBytes (byte[] bytes, int offset, int length, boolean append) {
 		return handles;
 	}
 
-	/** Returns true if this file is a directory. Always returns false for classpath files. On Android, an
-	 * {@link FileType#Internal} handle to an empty directory will return false. On the desktop, an {@link FileType#Internal}
-	 * handle to a directory on the classpath will return false. */
+	 /**
+	  * Returns true if this file is a directory. Always returns false for classpath files. On Android, an
+	  * {@link FileType#Internal} handle to an empty directory will return false unless you are using an asset index. On desktop,
+	  * an {@link FileType#Internal} handle will always return false unless you are using an asset index.
+	  *
+	  * See {@link #list()} documentation for more information about using an Asset Index
+	  */
 	public boolean isDirectory () {
-		if (type == FileType.Classpath) return false;
-		return file().isDirectory();
+		 if (type == FileType.Classpath)
+			  return false;
+		 if (type == FileType.Internal) {
+			  for (String indexpath : assetIndex()) {

```

Comment:
This should be cached. The internal dir is not going to change after first run, many calls to isDirectory will generate a lot of garbage.